clients/da-vinci-client/src/main/java/com/linkedin/davinci/bootstrap/DvcP2PRocksDbBootstrapper.java

```diff
@@ -0,0 +1,115 @@
+    String routerUrl = backend.getClientConfig().getVeniceURL();
+
+    this.p2pBlobTransferBaseDir = serverConfig.getDvcP2pBlobTransferBaseDir();
+    this.isDvcP2pBlobTransferEnabled = serverConfig.isDvcP2pBlobTransferEnabled();
+    this.veniceStoreVersionConfig = veniceStoreVersionConfig;
+    this.backend = backend;
+    this.isBlobTransferEnabled = isBlobTransferEnabled;
+    this.isHybridStore = isHybridStore;
+
+    P2PBlobTransferService blobTransferService =
+        new P2PBlobTransferService(p2pBlobTransferPort, p2pBlobTransferBaseDir);
+    NettyFileTransferClient fileTransferClient =
+        new NettyFileTransferClient(p2pFileTransferPort, p2pBlobTransferBaseDir);
+    TransportClient transportClient = ClientFactory.getTransportClient(backend.getClientConfig());
+    BlobFinder blobFinder = new DvcBlobFinder(transportClient, routerUrl);
+    this.p2pBlobTransferManager = new NettyP2PBlobTransferManager(blobTransferService, fileTransferClient, blobFinder);
+
+  }
+
+  /**
+   * Starts consuming data for the specified partition of the store and version.
+   * First, it calls the blob manager to transfer blobs from a peer node to a given path
+   * Then, it attempts to bootstrap the database from the given path
+   * If it fails to bootstrap from blobs at anypoint, it falls back to consuming data from Kafka
+   *
+   * @param storeName - store Name
+   * @param versionNumber - store version
+   * @param partitionId - partition ID for the store & version
+   * @throws VeniceBootstrapException
+   */
+  @Override
+  public void bootstrapDatabase(String storeName, int versionNumber, int partitionId) throws VeniceBootstrapException {
+    if (!isDvcP2pBlobTransferEnabled || !isBlobTransferEnabled || isHybridStore) {
+      bootstrapFromKafka(partitionId);
+    } else {
+      bootstrapFromBlobs(storeName, versionNumber, partitionId);
+    }
+  }
+
+  @Override
+  protected void bootstrapFromBlobs(String storeName, int versionNumber, int partitionId)
+      throws VeniceBootstrapException {
+    try {
+      p2pBlobTransferManager.start();
+
+      CompletionStage<InputStream> p2pTransfer = p2pBlobTransferManager.get(storeName, versionNumber, partitionId);
+      p2pTransfer
+          .whenComplete((inputStream, throwable) -> handleP2PTransferCompletion(inputStream, throwable, partitionId));
+    } catch (Exception e) {
+      LOGGER.error("Error occurred during bootstrap from blobs: {}", e.getMessage());
+      bootstrapFromKafka(partitionId);
+    }
+  }
+
+  private void handleP2PTransferCompletion(InputStream inputStream, Throwable throwable, int partitionId) {
+    if (throwable != null) {
+      LOGGER.error("Error occurred during p2p transfer: {}", throwable.getMessage());
+      bootstrapFromKafka(partitionId);
+      return;
+    }
+
+    try {
+      // P2P transfer is successful, attempt to bootstrap from blobs
+      verifyBootstrap(p2pBlobTransferBaseDir);
```

Comment:
When `verifyBootstrap` has some issues, we just do some loggings so the transfer is always mistakenly treated as "success". Is this true?